libyul/optimiser/UnusedStoreBase.cpp

```diff
@@ -37,47 +37,50 @@ void UnusedStoreBase::operator()(If const& _if)
 {
 	visit(*_if.condition);
 
-	TrackedStores skipBranch{m_stores};
+	ActiveStores skipBranch{m_activeStores};
 	(*this)(_if.body);
 
-	merge(m_stores, std::move(skipBranch));
+	merge(m_activeStores, std::move(skipBranch));
 }
 
 void UnusedStoreBase::operator()(Switch const& _switch)
 {
 	visit(*_switch.expression);
 
-	TrackedStores const preState{m_stores};
+	ActiveStores const preState{m_activeStores};
 
 	bool hasDefault = false;
-	vector<TrackedStores> branches;
+	vector<ActiveStores> branches;
 	for (auto const& c: _switch.cases)
 	{
 		if (!c.value)
 			hasDefault = true;
 		(*this)(c.body);
-		branches.emplace_back(std::move(m_stores));
-		m_stores = preState;
+		branches.emplace_back(std::move(m_activeStores));
+		m_activeStores = preState;
 	}
 
 	if (hasDefault)
 	{
-		m_stores = std::move(branches.back());
+		m_activeStores = std::move(branches.back());
 		branches.pop_back();
 	}
 	for (auto& branch: branches)
-		merge(m_stores, std::move(branch));
+		merge(m_activeStores, std::move(branch));
 }
 
 void UnusedStoreBase::operator()(FunctionDefinition const& _functionDefinition)
 {
-	ScopedSaveAndRestore outerAssignments(m_stores, {});
+	ScopedSaveAndRestore allStores(m_allStores, {});
+	ScopedSaveAndRestore usedStoresStores(m_usedStores, {});
+	ScopedSaveAndRestore outerAssignments(m_activeStores, {});
 	ScopedSaveAndRestore forLoopInfo(m_forLoopInfo, {});
 	ScopedSaveAndRestore forLoopNestingDepth(m_forLoopNestingDepth, 0);
 
 	(*this)(_functionDefinition.body);
 
 	finalizeFunctionDefinition(_functionDefinition);
+	m_storesToRemove += m_allStores - m_usedStores;
```

Comment:
```suggestion
	std::set_difference(
		m_allStores.begin(), m_allStores.end(),
		m_usedStores.begin(), m_usedStores.end(),
		std::inserter(m_storesToRemove, m_storesToRemove.end())
	);
```
This would be interesting to benchmark on your test case, as it should technically avoid copies; also, had no idea `std::set` supports `+`, `-`, etc. operators, and I can't find it anywhere in the docs?

Reply:
Right, `m_storesToRemove` is a `vector`, and I also tested it out and it doesn't seem to make a difference.